x-pack/plugins/ml/public/data_frame/components/aggregation_list/popover_form.tsx

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  EuiButton,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiSelect,
+} from '@elastic/eui';
+
+import { dictionaryToArray } from '../../../../common/types/common';
+
+import {
+  AggName,
+  isAggName,
+  PivotAggsConfig,
+  PivotAggsConfigDict,
+  PIVOT_SUPPORTED_AGGS,
+} from '../../common';
+
+interface SelectOption {
+  text: string;
+}
+
+interface Props {
+  defaultData: PivotAggsConfig;
+  otherAggNames: AggName[];
+  options: PivotAggsConfigDict;
+  onChange(d: PivotAggsConfig): void;
+}
+
+export const PopoverForm: React.SFC<Props> = ({
```

Comment:
It might be worth trying to widen the fields in the Popover, or going for a vertical layout, as the form field inputs are not very wide. E.g. with metricbeat fields:

![image](https://user-images.githubusercontent.com/7405507/56907701-bb389200-6a9c-11e9-8732-03a24297f914.png)

Reply:
I agree that's better, switched to vertical layout in 279b56e.

<img src="https://user-images.githubusercontent.com/230104/56911299-eaef9600-6aac-11e9-9b7a-4f54aa04d58d.png" width="50%" />